app/data/events/scheme-id/get-submitted-events.js

```diff
@@ -0,0 +1,18 @@
+const { getClient, odata } = require('../../../storage')
+const { PAYMENT_EVENT } = require('../../../constants/event-types')
+const { PAYMENT_SUBMITTED } = require('../../../constants/events')
+
+const getSubmittedEvents = async (category) => {
+  const client = getClient(PAYMENT_EVENT)
+  const eventResults = client.listEntities({ queryOptions: { filter: odata`category eq '${category}' and type eq '${PAYMENT_SUBMITTED}'` } })
```

Comment:
Could you explain what this line of code does?

Reply:
`client` is defined on line 6 and is getting the payment instance of the Azure Table Storage, this `client` has different functions we can call to interact with the data that is stored in the table.

line 7 is using the `listEntities` function to get all the 'entities' (which in our case are the events). We are then passing in an object which contains query data to filter the events. Specifically we are telling it to filter and only return the events that have a category of category (`schemeId`) and type of payment_submitted (`uk.gov.defra.ffc.pay.payment.submitted`)

So basically we use this line to get the events from the table storage based on the filters we pass in.

Reply:
Thanks!